### PR TITLE
Implement support for NDJSON streaming in vertex-web

### PR DESCRIPTION
**Pull request by ntrp**
I noticed Vertx reactive routes support SSE but not NDJSON (usually is the one that is used by spring webflux). Spring Webflus was returning `application/stream+json` before but they are now switching to `application/x-ndjson`, this is why I am using that content type.

https://github.com/spring-projects/spring-framework/issues/21283

If this integration is wanted I will polish and write test, otherwise I'll not waste more time on it, please let me know.

p.s. I did not touch the other supports but the the SSE one has a minor bug imho, the `initialize` function always applies the header since the headers are always empty. It is necessary to check the `rc.getAcceptableContentType()` method to verify if anybody set a custom produces value for the route. In case I can fix that too on the MultiSSESupport.

### CODE REVIEW

**Comment by cescoffier** [comment]
Interesting. Is there any document explaining NDJSON?

@geoand FYI.

File: extensions/vertx-web/deployment/src/main/java/io/quarkus/vertx/web/deployment/Methods.java [hunk 1]
```diff
@@ -26,12 +26,7 @@
 import io.quarkus.gizmo.FieldCreator;
 import io.quarkus.gizmo.MethodDescriptor;
 import io.quarkus.gizmo.ResultHandle;
-import io.quarkus.vertx.web.runtime.MultiJsonArraySupport;
-import io.quarkus.vertx.web.runtime.MultiSseSupport;
-import io.quarkus.vertx.web.runtime.MultiSupport;
-import io.quarkus.vertx.web.runtime.RouteHandler;
-import io.quarkus.vertx.web.runtime.RouteHandlers;
-import io.quarkus.vertx.web.runtime.ValidationSupport;
+import io.quarkus.vertx.web.runtime.*;
```
Comment (cescoffier) [inline extensions/vertx-web/deployment/src/main/java/io/quarkus/vertx/web/deployment/Methods.java:29]: Don't use start in imports.

**Review by cescoffier — CHANGES_REQUESTED** [review]
Can you add some tests?

**Comment by geoand** [comment]
@FroMage do you remember this one from your research around JSON streaming?

**Comment by FroMage** [comment]
I don't remember. But the issue was #13663

**Comment by ntrp** [comment]
I could not run the tests for some reason locally, not in Idea and also not with maven..
In Idea somehow all modules have been imported to java 8 instead of 11 even if I set the Porject jdk to 11. Diamond operators in other classes break because of that.

In Maven instead I get errors like this:
```
[ERROR] Failures:
[ERROR]   MultipleThrowableParamsFailureHandlerTest Build failed with wrong exception, expected class java.la
ng.IllegalStateException but got io.quarkus.builder.ChainBuildException: No producers for required item class
 io.quarkus.arc.deployment.TransformedAnnotationsBuildItem ==> expected: <true> but was: <false>
```
when I run `mvn test -pl :quarkus-vertx-web-deployment`

Any hints? In the meantime I wrote the tests and pushed hoping they are fine..

**Comment by geoand** [comment]
> when I run `mvn test -pl :quarkus-vertx-web-deployment`
> 
> Any hints? In the meantime I wrote the tests and pushed hoping they are fine..

Try building the entire project with: `mvn -T C1 -Dquickly` and the running the tests in the module with `mvn install -f extensions/vertx-web/`

**Comment by ntrp** [comment]
Thanks, that did the trick, now I can run the test via `mvn`. Still cannot run via Intellij due to the fact that the modules are set to Java 8 but better then nothing.

Do you have an idea on why the modules got imported as java 8 when in the pom I've seen it's set `maven.compiler.target` to `11`?

**Comment by geoand** [comment]
See https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/IDEA.20modules.20and.20java.208

**Review by cescoffier — APPROVED** [review]

**Comment by ntrp** [comment]
> See https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/IDEA.20modules.20and.20java.208

That worked, should we add it to the CONTRIBUTION.md? Either in the Q&A section or in the Intellij IDE setup one.

**Comment by geoand** [comment]
I added it this morning 🙂

**Comment by geoand** [comment]
@cescoffier is ready for merge?

**Pull request merged**